src/dos/drive_local.cpp

```diff
@@ -81,37 +87,47 @@ bool localDrive::FileOpen(DOS_File * * file,char * name,Bit32u flags) {
 
 	//Flush the buffer of handles for the same file. (Betrayal in Antara)
 	Bit8u i,drive=DOS_DRIVES;
-	localFile *lfp;
-	for (i=0;i<DOS_DRIVES;i++) {
-		if (Drives[i]==this) {
-			drive=i;
+	localFile *lfp = nullptr;
+	for (i = 0; i < DOS_DRIVES; i++) {
+		if (Drives[i] == this) {
+			drive = i;
 			break;
 		}
 	}
-	for (i=0;i<DOS_FILES;i++) {
+	for (i = 0; i < DOS_FILES; i++) {
 		if (Files[i] && Files[i]->IsOpen() && Files[i]->GetDrive()==drive && Files[i]->IsName(name)) {
 			lfp=dynamic_cast<localFile*>(Files[i]);
 			if (lfp) lfp->Flush();
 		}
 	}
 
-	FILE * hand = fopen_wrap(newname,type);
-//	Bit32u err=errno;
-	if (!hand) { 
-		if((flags&0xf) != OPEN_READ) {
-			FILE * hmm = fopen_wrap(newname,"rb");
-			if (hmm) {
-				fclose(hmm);
-				LOG_MSG("Warning: file %s exists and failed to open in write mode.\nPlease Remove write-protection",newname);
+	FILE* fhandle = fopen(newname, type);
+	// Was the file opened with the desired type?
+	if (!fhandle) {
+		// No; but do the requested flags require write-access?
+		if ((flags & 0xf) != OPEN_READ) {
```

Comment:
Out of curiosity, I took a peek at what flags exactly are we taking in / throwing out when doing `flags & 0xf`. I am not sure if this if works as it is supposed to…
```
enum {
OPEN_READ=0,
OPEN_WRITE=1,
OPEN_READWRITE=2,
OPEN_READ_NO_MOD=4,
DOS_NOT_INHERIT=128
};
```
I am not sure what `READ_NO_MOD` is for exactly… Should it be included in this if?

Reply:
Thanks for pointing this out. We actually only want to handle one exact situation: The DOS games tries `OPEN_READWRITE` _and_ the local file exists but is write-protected. All the other flags will be handled correctly.
- OPEN_READ -> Perfectly fine with a write-protected file.
- OPEN_WRITE -> We want this to fail, because the DOS program won't be reading anyway
- OPEN_READ_NO_MOD -> Perfectly fine with a write-protected file
- DOS_NOT_INHERIT=128 ? I'm not sure what this is.